pkg/sparrow/run.go

```diff
@@ -49,16 +49,15 @@ type Sparrow struct {
 	checkCoordinator
 }
 
+// checkCoordinator is used to coordinate the checks and the reconciler
 type checkCoordinator struct {
```

Comment:
do we really still need this struct? Can't the Controller replace it completely. It seems like an unneeded indirection.